### DISCUSSION TITLE
pdo_odbc driver for Doctrine possible? Unconventional database server.

### DISCUSSION THREAD

**Discussion by techsnek** [symfony/symfony#50058] [Q&A] [answered]
My understanding is that there is no compatibility for the pdo_odbc driver for Doctrine. Is this correct? I am using an unconventional database type unfortunately (Advantage Database Server - ADS SAP), and I have to use either baseline odbc or pdo_odbc to interface with the database from PHP. Is there any way that I would be able to use Doctrine through pdo_odbc somehow or am I just flat out of luck? Are there any unofficial tools I can use perhaps? Symfony is perfect in every conceivable way for my current project aside from this database issue so I'm hoping something can be figured out, even if not ideal. 

Any help is appreciated, thanks.

**Comment by mbabker** [comment, marked as answer]
You'll need to roll your own code for that PDO driver since Doctrine doesn't natively support it.  https://www.doctrine-project.org/projects/doctrine-dbal/en/current/reference/supporting-other-databases.html covers that on a very high level.

You'd be able to use some of the existing PDO classes for this integration, but it looks like the most complex part is in the code you'll have to write, namely the schema manager and platform classes.

**Reply by techsnek** [reply]
Thank you for the help. Looks like I have quite a bit of Doctrine driver coding to do. I was hoping that with the other PDO drivers being available for use with Doctrine that there would be some form of ODBC/PDO_ODBC implementation.
